ExampleMod/Common/Players/ExampleFishingPlayer.cs

```diff
@@ -63,8 +63,26 @@ public class ExampleFishingPlayer : ModPlayer
 				// Let's replace biome crates 50% of the time (player could be in multiple (modded) biomes, we should respect that)
 				if (!attempt.veryrare && !attempt.legendary && attempt.rare && Main.rand.NextBool()) {
 					itemDrop = ModContent.ItemType<Content.Items.Consumables.ExampleFishingCrate>();
+					return; // This is important so your code after this that rolls items will not run
 				}
 			}
+
+			// Here we will set the catch conditions for our ExampleQuestFish
+			int exampleQuestFish = ModContent.ItemType<Content.Items.ExampleQuestFish>(); // We'll store the type as a variable, since we'll be referencing it several times
+			// First we check if today's quest matches our quest fish
+			if (attempt.questFish == ModContent.ItemType<Content.Items.ExampleQuestFish>()) {
+				// Now we'll check if the player already has the fish, and if they've completed today's quest yet
+				if (!Player.HasItem(exampleQuestFish) && !Main.anglerQuestFinished) {
```

Comment:
This seems to be a regression from 1.3... Does vanilla not check for these two conditions anymore (checking `attempt.questFish`)?

Reply:
You are right, vanilla does account for this. I misunderstood the code and have made a commit to remove the extra checks.
The 1.3 code was at some point removed. This PR exists because the catching of exampleQuestFish was impossible.